source/utils/winston.js

```diff
@@ -0,0 +1,24 @@
+const winston = require('winston');
+
+const consoleTransport = winston.default.transports.find((transport) => {
+  return transport instanceof winston.transports.Console;
+});
+if (!consoleTransport) {
+  winston.add(new winston.transports.Console({
+    level: "error",
+    format: winston.format.combine(
+      winston.format.timestamp(),
+      winston.format.colorize(),
+      winston.format.printf((info) => {
+        const splat = info[Symbol.for("splat")];
+        if (splat) {
+          const splatStr = splat.map((arg) => JSON.stringify(arg)).join("\n");
+          return `${info.timestamp} - ${info.level}: ${info.message} ${splatStr}`;
+        }
+        return `${info.timestamp} - ${info.level}: ${info.message}`;
+      })
+    )
+  }));
+}
+
+module.exports = winston;
```

Comment:
It feels a bit weird to have this file be imported/required everywhere seemingly for its side effects.
Is there not a better way to ensure the logging is set up correctly?

Reply:
I have changed it so it is only required by ungits entrypoints which are:
- `bin/credentials-helper`
- `bin/ungit`
- `nmclicktests/environment.js`
- `public/main.js`
- `source/server.js`

With winston 3 there is no default console transport that's why I did it this way. Also if no transport is specified winston will complain about that every time something is logged.